internal/engine/image_build_and_deployer.go

```diff
@@ -33,15 +33,15 @@ func NewImageBuildAndDeployer(b build.ImageBuilder, k8sClient k8s.Client, histor
 	}, nil
 }
 
-func (ibd ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, service model.Service, state BuildState) (BuildResult, error) {
+func (ibd ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, service model.Service, state BuildState) (br BuildResult, err error) {
```

Comment:
Why are we using named returns here now?

Reply:
cuz we want the deferred call to `EndPipeline` to report on the state of the error that the function is returning. without named returns, we have no way to latch onto the value we're returning, and rely on it having been assigned to a variable beforehand (and not just any variable, but a specific one -- so users can't name multiple errors different things).
```
func Stuff() error {
    var err error
    defer EndPipeline(err)
    ...
    return fmt.Errorf("this error isn't bound to 'err'"!)
```
^ if we don't use named returns, we have no way of grabbing that error we return here.